### DISCUSSION TITLE
SX1262 crashes during RadioIrqProcess

### DISCUSSION THREAD

**Discussion by ghalencar** [Lora-net/LoRaMac-node#1238] [Q&A] [answered]
Good morning,

I need help. This doesn't happen often, but I wouldn't want you to reset my part. When a message is received it always passes through the function "void RadioIrqProcess( void )" , but when the message arrives in the following decisions:
IRQ_CRC_ERROR, IRQ_PREAMBLE_DETECTED and IRQ_HEADER_ERROR the system crashes and restarts via an external watchdog. Is there anything that can be done in this function?

**Comment by mluis1** [comment, marked as answer]
Could you check that all `RadioEvents_t` callbacks are properly initialized?

Do you observe hard fault interrupts? 
If yes, most of the time it means that some callback are not properly initialized. However this should not happen as all callbacks execution are protected.

Examples:
`IRQ_CRC_ERROR`
https://github.com/Lora-net/LoRaMac-node/blob/fe8247e2b84101fe701531a5f9ef14f035743af4/src/radio/sx126x/radio.c#L1293-L1296

`IRQ_PREAMBLE_DETECTED` no code is executed when this happens
https://github.com/Lora-net/LoRaMac-node/blob/fe8247e2b84101fe701531a5f9ef14f035743af4/src/radio/sx126x/radio.c#L1355-L1358

`IRQ_HEADER_ERROR`
https://github.com/Lora-net/LoRaMac-node/blob/fe8247e2b84101fe701531a5f9ef14f035743af4/src/radio/sx126x/radio.c#L1378-L1381

`RadioIrqProcess`
https://github.com/Lora-net/LoRaMac-node/blob/fe8247e2b84101fe701531a5f9ef14f035743af4/src/apps/LoRaMac/common/LmHandler/LmHandler.c#L354-L358

In order to debug your issue maybe you can launch a debug session and set a break point at first statement inside the concerned `if` statement then execute the code step by step up until you observe the issue.
Once the observed issue is understood you may find soluttions on how to solve it.

**Comment by ghalencar** [comment]
Thanks for the answer!

I managed to solve. I put a flag that only calls the following function when lorawan starts:

void RadioOnDioIrq( void )
{
     IrqFired = true;
     if(st_LoraWan.bLWConfigOK == true)
       RadioIrqProcess();
}

Also when preamble or CRC error occurred, I reset the radio by pin, put a timeout on the SPI and restarted Lorawan.

After that, I had no more problems.
